pool/upgrades_test.go

```diff
@@ -110,42 +115,55 @@ func verifyV2Upgrade(t *testing.T, db *bolt.DB) {
 }
 
 func verifyV3Upgrade(t *testing.T, db *bolt.DB) {
+	funcName := "verifyV3Upgrade"
 	err := db.View(func(tx *bolt.Tx) error {
 		pbkt := tx.Bucket(poolBkt)
 		if pbkt == nil {
-			desc := fmt.Sprintf("bucket %s not found", string(poolBkt))
-			return MakeError(ErrBucketNotFound, desc, nil)
+			desc := fmt.Sprintf("%s: bucket %s not found",
+				funcName, string(poolBkt))
+			return dbError(ErrBucketNotFound, desc)
 		}
 
 		sbkt := pbkt.Bucket(paymentBkt)
 		if sbkt == nil {
-			desc := fmt.Sprintf("bucket %s not found", string(paymentBkt))
-			return MakeError(ErrBucketNotFound, desc, nil)
+			desc := fmt.Sprintf("%s: bucket %s not found",
+				funcName, string(paymentBkt))
+			return dbError(ErrBucketNotFound, desc)
 		}
 
 		c := sbkt.Cursor()
 		for k, v := c.First(); k != nil; k, v = c.Next() {
 			var payment Payment
 			err := json.Unmarshal(v, &payment)
 			if err != nil {
-				return err
+				desc := fmt.Sprintf("%s: unable to unmarshal payment: %v",
+					funcName, err)
+				return dbError(ErrParse, desc)
 			}
 
-			id := paymentID(payment.Height, payment.CreatedOn, payment.Account)
+			id, err := paymentID(payment.Height, payment.CreatedOn, payment.Account)
+			if err != nil {
+				return err
+			}
 			if !bytes.Equal(k, id) {
-				return fmt.Errorf("expected payment id (%x) to be the same as "+
-					"its key (%x)", id, k)
+				desc := fmt.Sprintf("%s: expected payment id (%x) to be "+
+					"the same as its key (%x)", funcName, id, k)
+				return dbError(ErrID, desc)
+
 			}
 
 			if payment.Source == nil {
-				return fmt.Errorf("expected a non-nil payment source")
+				desc := fmt.Sprintf("%s: expected a non-nil "+
```

Comment:
`err` is always going to be `nil` at this point

Reply:
`err` isn't passed for `desc` here, not sure what the issue here is, please do elaborate.

Reply:
![Screenshot from 2020-09-15 11-00-30](https://user-images.githubusercontent.com/6762864/93196906-e7f95200-f742-11ea-93a0-8026578aae8d.png)